components/sdmmc/sdmmc_cmd.c

```diff
@@ -204,135 +215,328 @@ esp_err_t sdmmc_card_init(const sdmmc_host_t* config, sdmmc_card_t* card)
         }
     }
 
-    /* Get the contents of SCR register: bus width and the version of SD spec
-     * supported by the card.
-     * In SD mode, this is the first command which uses D0 line. Errors at
-     * this step usually indicate connection issue or lack of pull-up resistor.
-     */
-    err = sdmmc_send_cmd_send_scr(card, &card->scr);
-    if (err != ESP_OK) {
-        ESP_LOGE(TAG, "%s: send_scr (1) returned 0x%x", __func__, err);
-        return err;
-    }
+    if (card->host.flags & SDMMC_HOST_MMC_CARD) {
+        /* sdmmc_mem_mmc_init */
+        int width, value;
+        int card_type;
+        uint8_t ext_csd[512];
```

Comment:
We try to avoid large allocations on the stack, if possible. Suggest using `heap_alloc_caps` instead.